src/python/pants/help/help_formatter.py

```diff
@@ -49,6 +52,13 @@ def add_option(ohis, *, category=None):
                 lines.extend([*self.format_option(ohi), ""])
 
         add_option(oshi.basic)
+        if oshi.advanced and not self._show_advanced:
+            lines.append(
+                self.maybe_green(
+                    f"There are advanced options which you can list by running "
+                    f"`./pants help-advanced {self.entity}`"
+                )
+            )
```

Comment:
I would prefer this block last (after the show deprecated block), otherwise, you'll get: `regular options -> this notice -> deprecated options` in case there are deprecated options.

Reply:
Sure, I've moved the block.